internal/controllers/clusterrequest_controller.go

```diff
@@ -1892,59 +1903,89 @@ func (t *clusterRequestReconcilerTask) updateNodeStatusWithHostname(ctx context.
 	return true
 }
 
-// updateHardwareProvisioningStatus updates the status for the created ClusterInstance
+// updateHardwareProvisioningStatus updates the status for the ClusterRequest
 func (t *clusterRequestReconcilerTask) updateHardwareProvisioningStatus(
-	ctx context.Context, nodePool *hwv1alpha1.NodePool) error {
+	ctx context.Context, nodePool *hwv1alpha1.NodePool) (bool, bool, error) {
```

Comment:
Optional: would it help using the same method as in https://github.com/openshift-kni/oran-o2ims/pull/210 to expose the transitions?
![image](https://github.com/user-attachments/assets/4eebf1f1-8c17-4fd9-9605-c41462b6e0b1)

Reply:
Thank you for your suggestions. I believe using an FSM would be overkill for this scenario. The current approach is simple and sufficient for the logic I’m implementing. Introducing an FSM would add unnecessary complexity and is more suited for handling complex transitions between multiple states. The current solution should work well while keeping the code clean.

Reply:
Ok got it, in this case it might be simple enough to follow. Personally I find it easier to extract the logic from the code (even in this case). In my opinion, it makes it easier to test independently, understand and share. Also, for any similar timeout we could easily reuse the FSM code/testing